app/src/test/java/com/naxtlevelofandroiddevelopment/marvelgallery/MainPresenterUnitTest.kt

```diff
@@ -0,0 +1,65 @@
+@file:Suppress("IllegalIdentifier")
+
+package com.naxtlevelofandroiddevelopment.marvelgallery
+
+import com.naxtlevelofandroiddevelopment.marvelgallery.data.network.Rx
+import com.naxtlevelofandroiddevelopment.marvelgallery.model.MarvelCharacter
+import com.naxtlevelofandroiddevelopment.marvelgallery.presenter.MainPresenter
+import com.nextlevelofandroiddevelopment.marvelgallery.helpers.Example.exampleCharacterList
+import com.nextlevelofandroiddevelopment.marvelgallery.helpers.MainViewHelper
+import com.nextlevelofandroiddevelopment.marvelgallery.helpers.MarvelRepositoryHelper
+import io.reactivex.Single
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+
+class MainPresenterUnitTest {
+
+    @Before
+    fun setUp() {
+        Rx.unitTestMode = true
+    }
+
+    @Test fun `After start, there is request for with null query`() {
+        checkSearchQuery(null) { onViewCreated() }
+    }
+
+    @Test fun `For blank text, there is request with null query`() {
+        checkSearchQuery(null) { onSearchChanged("") }
```

Comment:
`checkSearchQuery(null) { onSearchChanged("") }` this is not readable at all, i wouldn't know that this method does without reading test name. It should be more like `assertOnAction ({ onSearchChanged("") }).thenSearchQueryIsEqualTo(null)`